src/commands/functions/create.js

```diff
@@ -301,12 +303,50 @@ const downloadFromURL = async function (context, flags, args, functionsDir) {
   }
 }
 
-const installDeps = function (functionPath) {
-  return new Promise((resolve) => {
-    cp.exec('npm i', { cwd: path.join(functionPath) }, () => {
-      resolve()
-    })
-  })
+// Takes an existing list of packages and a list of packages needed by a
+// function, and computes the list of npm modules with version range that
+// must be installed.
+const getNpmInstallPackages = (existingPackages = {}, neededPackages = {}) =>
+  Object.entries(neededPackages)
+    .filter(([name]) => existingPackages[name] === undefined)
+    .map(([name, version]) => `${name}@${version}`)
+
+// When installing a function's dependencies, we first try to find a site-level
+// `package.json` file. If we do, we look for any dependencies of the function
+// that aren't already listed as dependencies of the site and install them. If
+// we don't do this check, we may be upgrading the version of a module used in
+// another part of the project, which we don't want to do.
+const installDeps = async ({ functionPackageJson, functionPath, functionsDir }) => {
+  // eslint-disable-next-line import/no-dynamic-require, node/global-require
+  const { dependencies: functionDependencies, devDependencies: functionDevDependencies } = require(functionPackageJson)
+  const sitePackageJson = await findUp('package.json', { cwd: functionsDir, type: 'file' })
```

Comment:
[sand] `type: 'file'` is the default value, so might be omitted.

Reply:
I personally like to be explicit, but I can remove if you feel strongly about it.

Reply:
Either works :+1:

Reply:
Actually, done in https://github.com/netlify/cli/pull/3125/commits/3aec5a7f41f3ea488e90d5634e8644cf1bd40133.